core/src/main/scala/com/rallyhealth/vapors/core/algebra/Expr.scala

```diff
@@ -190,6 +191,15 @@ object Expr {
     override def visit[G[_]](v: Visitor[V, P, G]): G[R] = v.visitUsingDefinitions(this)
   }
 
+  final case class CustomFunction[V, A : Typeable, R, P](
+    inputExpr: Expr[V, A, P],
+    name: String,
+    evaluate: A => R,
+    capture: CaptureP[V, R, P],
+  ) extends Expr[V, R, P] { // TODO: Should this return a Try[R]? Either[Failure, R]?
```

Comment:
I created this ticket for follow-up: https://jira.rallyhealth.com/browse/PRO-1040